tsconfig.json

```diff
@@ -66,8 +66,8 @@
 
     /* Advanced Options */
     "skipLibCheck": true,                           /* Skip type checking of declaration files. */
-    "forceConsistentCasingInFileNames": true        /* Disallow inconsistently-cased references to the same file. */
+    "forceConsistentCasingInFileNames": true,        /* Disallow inconsistently-cased references to the same file. */
   },
   "include": ["src"],
-  "exclude": ["node_modules", "**/testing/*", "**/*.spec.ts"]
```

Comment:
i think what this line doing is excluding the testing files from test.

Reply:
yes, we need to use node package library on spec.ts file. if we exclude it, it means spec.ts files cannot use what we `npm install`-ed in the package.json.

Reply:
please also look at the "npm run test" or "npm run publish" commands, those works by building with test scripts excluded with "tsconfig-build.json"

Reply:
@sasangachathumal it's more proper to develop with tsconfig.json, and do deployment + testing the final js bundle with this https://github.com/intersective/practera-sdk/pull/7/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L24

Reply:
@trtshen if you check develop branch,
it still have this exclude in tsconfig.json. https://github.com/intersective/practera-sdk/blob/develop/tsconfig.json#L72
and is spec.ts files i have use lodash, axios in test. https://github.com/intersective/practera-sdk/blob/develop/src/testing/request-service.spec.ts
so is that wrong. or i misunderstand what you mention.

Reply:
@sasangachathumal  yup, that's not so proper way of using ts-jest.
Since we are depending on ts-jest (and we have jest.config.js), your spyOn works for the tests script you wrote, but for more advanced use of jest, we cant "exclude" spec.ts from tsconfig.

Reply:
@sasangachathumal you can check how the jest being used to mock a module, and clear its cache if we have tests wrote for similar function.
https://github.com/intersective/practera-sdk/pull/7/files#diff-840bc865529e6bb6934d078299b231f109d7db04bfedf6abc0202e91756c59b3R7

Reply:
ok. cool. thanks for clarification. then we can mock modules in  test file. so we need to update other spec.ts files according to that.